schemas/testpilot.js

```diff
@@ -1,21 +1,40 @@
 "use strict";
 
-const Joi = require("joi");
+const Joi = require("joi-browser");
 const commonSchema = require("./commonSchema");
 
-const testpilotEvent = Joi.object().keys({
-  event: Joi.string().required(),
-  object: Joi.string().required(),
-  timestamp: Joi.number().integer().required()
-});
-
+// commonSchema keys not shown below:
+// - event_type, called "event" in testpilot docs, and "method" in
+//   testpilot-metrics docs;
+// - client_id, called "clientId" in the Telemetry ping.
 const schema = commonSchema.keys({
-  payload: Joi.object().required().keys({
-    events: Joi.array().required().length(1).items(testpilotEvent),
-    test: Joi.string().valid("@testpilot-addon").required(),
-    timestamp: Joi.number().integer().required(),
-    version: Joi.string().required()
-  })
-}).options({allowUnknown: true});
+  // Require that the commonSchema "topic" field always be "testpilot".
+  topic: Joi.string().valid("testpilot").required(),
+
+  // Additional "testpilot" event fields:
+  // object: what was affected by "event", e.g., a button.
+  object: Joi.string(),
+
+  // client_time: non-PII timestamp, integer number of seconds since Firefox
+  //              application start.
+  client_time: Joi.number().integer().required(),
+  // variants: an optional string (either a simple string, or serialized JSON)
+  //           that describes any A/B or multivariate tests currently running.
+  //           See the variant docs in the mozilla/testpilot repo for more.
+  variants: Joi.string(),
```

Comment:
@emtwo great question.

The idea of an arbitrary `variants` object doesn't really work with Google Analytics, so, for new experiments, the suggestion is to make the first GA Custom Dimension a 'variant' string field, and use some kind of simple string to identify both the experiment and the experimental group for a given user. This is documented in [the testpilot-metrics library](https://github.com/mozilla/testpilot-metrics#readme), and the not-yet-merged changes to the [testpilot metrics docs](https://github.com/mozilla/testpilot/pull/1662). 

For the case of existing addons, we'll just make sure to double-serialize the `variants` field, so that it's a JSON string, even after it's deserialized by onyx. Same idea as the double-serialized `raw` field in the new ping-centre `testpilot` schema: easy to store in Redshift for the simple default case, and workable for older addons or more complex addons.

Now that I'm writing all this down, I think experiment authors are going to need some updated examples of how to work with variants. I'll add that to my docs todo list.